static/app/views/replays/detail/console/index.tsx

```diff
@@ -10,24 +16,51 @@ interface Props {
 }
 
 function Console({breadcrumbs}: Props) {
+  const [searchTerm, setSearchTerm] = useState('');
+  const handleSearch = debounce(query => setSearchTerm(query), 150);
+  const filteredBreadcrumbs = useMemo(
+    () =>
+      !searchTerm
+        ? breadcrumbs
+        : breadcrumbs.filter(breadcrumb =>
+            breadcrumb.message?.toLowerCase().includes(searchTerm)
+          ),
+    [searchTerm, breadcrumbs]
+  );
   return (
-    <ConsoleTable>
-      {breadcrumbs.map((breadcrumb, i) => (
-        <ConsoleMessage
-          key={i}
-          isLast={i === breadcrumbs.length - 1}
-          breadcrumb={breadcrumb}
-        />
-      ))}
-    </ConsoleTable>
+    <Fragment>
+      <StyledSearchBar onChange={handleSearch} />
+      {filteredBreadcrumbs.length > 0 ? (
+        <ConsoleTable>
+          {filteredBreadcrumbs.map((breadcrumb, i) => (
+            <ConsoleMessage
+              key={i}
+              isLast={i === breadcrumbs.length - 1}
+              breadcrumb={breadcrumb}
+            />
+          ))}
+        </ConsoleTable>
+      ) : (
+        <StyledEmptyMessage title={t('No results found.')} />
+      )}
+    </Fragment>
   );
 }
 
+const StyledEmptyMessage = styled(EmptyMessage)`
+  align-items: center;
+`;
+
 const ConsoleTable = styled(Panel)`
   display: grid;
   grid-template-columns: max-content auto;
   font-family: ${p => p.theme.text.familyMono};
   font-size: 0.8em;
 `;
 
+const StyledSearchBar = styled(SearchBar)`
```

Comment:
Can we try to avoid the use of `Styled` as a prefix? Something like `<ConsoleSearch>` would be better

Reply:
I'm good with that, I'll adjust it.